src/edgepi/pwm/edgepi_pwm.py

```diff
@@ -41,104 +39,217 @@ def __check_range(self, target, range_min, range_max) -> bool:
+    def get_duty_cycle(self, pwm_num: PWMPins):
         """
         Get duty_cycle
         Args:
-            N/A
+            pwm_num (PWMPins): target pwm device
         Returns:
             duty_cycle (int): duty_cycle value in percentage
         """
-        return int(self.pwm.get_duty_cycle_pwm() * 100)
+        if pwm_num is None:
+            raise ValueError(f"get_duty_cycle: PWM number is missing {pwm_num}")
+        return self.__pwm_1.get_duty_cycle_pwm()*100 if pwm_num == PWMPins.PWM1 else\
+               self.__pwm_2.get_duty_cycle_pwm()*100
 
-    def set_polarity(self, polarity: Polarity):
+    def __set_polarity(self, pwm_num: PWMPins, polarity: Polarity):
         """
         Set polarity
         Args:
+            pwm_num (PWMPins): target pwm device
             polarity (int): polarity value
         Returns:
             N/A
         """
-        self.pwm.set_polarity_pwm(polarity.value)
+        if pwm_num is None:
+            raise ValueError(f"__set_polarity: PWM number is missing {pwm_num}")
+        if pwm_num == PWMPins.PWM1:
+            self.__pwm_1.set_polarity_pwm(polarity.value)
+        else:
+            self.__pwm_2.set_polarity_pwm(polarity.value)
 
-    def get_polarity(self):
+
+    def get_polarity(self, pwm_num: PWMPins):
         """
         Get polarity
         Args:
-            N/A
+            pwm_num (PWMPins): target pwm device
         Returns:
             polarity (int): polarity value
         """
-        return self.pwm.get_polarity_pwm()
+        if pwm_num is None:
+            raise ValueError(f"get_polarity: PWM number is missing {pwm_num}")
+        return self.__pwm_1.get_polarity_pwm() if pwm_num == PWMPins.PWM1 else\
+               self.__pwm_2.get_polarity_pwm()
+
 
-    def enable(self):
+    def enable(self, pwm_num: PWMPins):
         """
         Enable pwm output
+        Args:
+            pwm_num (PWMPins): target pwm device
+        Returns:
+            N/A
         """
-        self.gpio.set_pin_state(GpioPins.AO_EN1.value if self.pwm_num==PWMPins.PWM1 else\
+        if pwm_num is None:
+            raise ValueError(f"enable: PWM number is missing {pwm_num}")
+        if self.get_enabled(pwm_num):
+            self.disable(pwm_num)
+        self.gpio.set_pin_state(GpioPins.AO_EN1.value if pwm_num==PWMPins.PWM1 else\
                                 GpioPins.AO_EN2.value)
-        self.gpio.clear_pin_state(GpioPins.DOUT1.value if self.pwm_num==PWMPins.PWM1 else\
+        self.gpio.clear_pin_state(GpioPins.DOUT1.value if pwm_num==PWMPins.PWM1 else\
                                 GpioPins.DOUT2.value)
-        self.gpio.clear_pin_state(self.pwm_num.value)
-        self.log.info("Enabling PWM")
-        self.pwm.enable_pwm()
+        self.gpio.clear_pin_state(pwm_num.value)
+        self.log.info("enable: Enabling PWM")
+        if pwm_num == PWMPins.PWM1:
+            self.__pwm_1.enable_pwm()
+        else:
+            self.__pwm_2.enable_pwm()
```

Comment:
sine you are using this if-else construct a lot in this file, I would recommend one of the following:
1. instead of two variables for pwm, use a dict like __pwm_devs then all this if-else will turn into:
```py
   self.__pwm_devs[pwm_num].enable_pwm()
```
2. create a func like `__get_pwm_dev` and put the if-else there and always call that func:
```py
   self.__get_pwm_dev(pwm_num).enable_pwm()
```
I prefer option 1  :D 
```